### ISSUE TITLE
portuguese translation

### ISSUE THREAD

**Issue by nafergo** [dainst/idai-field#66]
Is it useful to do a portuguese translation? If so, can you point how to proceed?

**Comment by tkleinke** [comment]
Of course we'd be happy to add a Portuguese translation to the application. You'd need to do the initial translation of the current texts and update the translation in future versions (when texts are edited or new texts are added). I just created a new branch "portuguese" and added translation files for the Portuguese version (which currently contain the English texts). You'd have to translate the following files:
 
["messages.pt.xlf"](https://github.com/dainst/idai-field/blob/portuguese/desktop/src/app/i18n/messages.pt.xlf): This file contains most of the texts shown in the app. You'd have to replace the English texts (inside of the "target" elements) with the Portuguese translations. You can either use a XLIFF translation software for editing the file or edit it directly in a text editor.

["messages.js"](https://github.com/dainst/idai-field/blob/portuguese/desktop/electron/messages.js), ["initialization-messages.ts"](https://github.com/dainst/idai-field/blob/portuguese/desktop/src/app/components/initialization-messages.ts), ["sample-data-labels.ts"](https://github.com/dainst/idai-field/blob/portuguese/core/src/datastore/sampledata/sample-data-labels.ts): In these files, the English texts in the section "pt" have to be replaced with the Portuguese translations.

["Core/Language.pt.json"](https://github.com/dainst/idai-field/blob/portuguese/core/config/Core/Language.pt.json), ["Library/Language.pt.json"](https://github.com/dainst/idai-field/blob/portuguese/core/config/Library/Language.pt.json): In these files, the English texts for "label", "description" and the different groups have to be replaced with the Portuguese translations.

If you decide to do the translation, it's probably easiest to create a fork, edit the files in the branch "portuguese" and create a pull request. This would also work for future updates.

We'd also need a contact mail address (for any questions and suggestions about the Portuguese translation) which will be shown in the language selection menu.

**Comment by poloarqueologicodeviseu** [comment]
I'll do it :) really interested in this :+1:

**Comment by nafergo** [comment]
I'm presently working with @poloarqueologicodeviseu and we'll do it :) I'm really interested in this 👍

**Comment by nafergo** [comment]
Core/Language.pt.json and messages.js are done. Should I create a pull request only after everything is translated or can I create a pull request with these two?
also, would it be possible to test an appimage with these files so that I can check everything inside running Field Desktop?

**Comment by tkleinke** [comment]
Yes, if you want to test the application with the current state of the translation, you can create a pull request (with branch "portuguese" as target). I'll trigger the build process and send you a download link for the AppImage file.

**Comment by poloarqueologicodeviseu** [comment]
I think (and hope) I did everything right... :) pls let us 
meanwhile, will continue to extend translation.

**Comment by tkleinke** [comment]
Great, everything looks fine. You can download a Linux build with the current state of the translation [here](https://cumulus.dainst.org/index.php/s/JJQFwnceJQf27w4).

**Comment by tkleinke** [comment]
Are you still working on the translation or do you need some additional information? We'd be happy to include the Portuguese translation in an official release once it's complete.

**Comment by nafergo** [comment]
Hi @tkleinke I'm working on it, yes. I'm waiting for some additional feedback from a fellow archaeologist about some labels. by the end of the next week I'll do a pull request with present translation. planning a new version?

**Comment by tkleinke** [comment]
Hi, that's great to hear. The next version 3.1 will probably be released in 1-2 weeks, but we can add the Portuguese translation in a later version, of course.  I just wanted to make sure that you're not waiting for any information from us.

**Comment by tkleinke** [comment]
@nafergo We are now managing translations with [Weblate](https://weblate.dainst.org) and have added the translations you already provided to the Weblate project. If you're still interested in contributing to the Portuguese translation, you can find all information in the [Wiki](https://github.com/dainst/idai-field/wiki/Translating-Field-Desktop).

**Comment by luna-watkins** [comment]
Hej. How is the status of the portuguese translation? I can contribute if needed. 
Greetings. 
Luna

**Comment by luna-watkins** [comment]
Ok, I made a Weblate-account (@lunawatkins).
Further languages I can contribute are English and German.

**Comment by luna-watkins** [comment]
Thanks!

**Comment by tkleinke** [comment]
Hi @luna-watkins, the Portuguese translation is planned to be released with the upcoming version 3.5. Could you please add the missing translations (a few new texts have also been added lately) in Weblate until January 27th if possible? The default valuelists do not have to be translated completely, but it would be great if you could add the missing translations for the configuration and the user interface.

**Comment by luna-watkins** [comment]
Hello @tkleinke. Then I will focus on config and interface. I will let you know when these are completed.

**Comment by luna-watkins** [comment]
Hi @tkleinke. I managed to finish the translation for everything except the valuelists.
I haven't had the time to make a full review of the translation (especially new orthography vs. old orthography, which is what I learned in school). To when should the valuelists be completely translated?

**Comment by tkleinke** [comment]
Great, thank you! There's no deadline for translating the default valuelists, in most languages they aren't translated completely yet. It would be great of course to have translations for them as well, but they are not mandatory for a new language to be added.

We're currently still waiting for other translations and will probably release the new version in early February. So if you do the review until then, we could still add the changes, otherwise the translation would be updated with the first patch release.

**Comment by luna-watkins** [comment]
Hey @tkleinke ,
I reviewed, unified and corrected everything from the Portuguese translation - it can be uploaded. 
Missing is only the _Default valuelists_, I will keep on doing it according to my availability. Please let me know when a deadline for this comes up down the road.
I encountered two issues using Weblate, which I would like to report:
**1 - Screenshot XMLMarkupNichtAngezeigt.JPG**
I have corrected all the XML markups shown to me, but I saw many that have not been marked up. They are identifiable by not having a blue highlight. I have NOT corrected them.
**2 - Screenshot alsUebersetztGeltend.JPG**
I don't know where the translation made by the account "None" came from, but most seem coming from an automatic translation. This mislead me by showing them to me as "translated". I have corrected them while doing the review. But perhaps automatically translated text should be marked as "needs editing" so a human can then double-check it.
Greetings,
Luna

![Image](https://github.com/user-attachments/assets/5853bb15-039a-485d-a504-6b8c16fb56fd)

![Image](https://github.com/user-attachments/assets/f359bdff-fd1e-460b-adc8-2640f33024a6)

**Comment by tkleinke** [comment]
Hi Luna, 

1. That should not be an issue. There's a known bug in the Angular library that we use which leads to wrong values in the "equiv-text" attributes of the XML elements. That's why these XML elements have changed in a lot of texts. But the attribute isn't used by the application, so it shouldn't matter if it's updated in the translation or not, as long as the correct order of elements (with the correct "id" attribute) is preserved.

2. These translations were provided by Nelson Gonçalves (see the discussion at the beginning of this issue). We didn't have Weblate back then, that's why there's no user name shown for his entries. I added his translations manually to Weblate.

I'm closing the issue now, if there's more questions about the translation or Weblate coming up, you can just create a new one. And if you want to adjust some translation texts in the future, you can always do that in Weblate. The changes will then automatically be included in the next release of the application. Many thanks again! :)
